lang/cs/Org.Apache.REEF.Utilities/Collections/PriorityQueue.cs

```diff
@@ -0,0 +1,225 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Utilities.Collections
+{
+    /// <summary>
+    /// A simple priority queue implementation, where the head of the queue is the 
+    /// smallest element in the priority queue.
+    /// </summary>
+    public sealed class PriorityQueue<T> : ICollection<T> where T : IComparable<T>
```

Comment:
The `GetHashCode` method in `Time` because it's a bizarre workaround to allow multiple instances of the same object in `SortedSet`. Using a `PriorityQueue` allows us to properly impose order on the `Time` objects.